tex/pgfcircshapes.tex

```diff
@@ -92,6 +92,23 @@
 		}
 }
 
+%% Diamond terminal
+
+\pgfdeclareshape{diamond}{
+	\anchor{center}{
+		\pgfpointorigin
+	}
+
+	\behindforegroundpath{		
+		\pgfscope
+			\pgftransformrotate{45}
+			\pgfpathrectanglecorners{\pgfpoint{\pgf@x+\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}}{\pgfpoint{\pgf@x-\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}{\pgf@y+\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2}}
```

Comment:
All right, that looks very ugly.
Can `\pgfkeysvalueof{/tikz/circuitikz/nodes width}*\pgfkeysvalueof{/tikz/circuitikz/bipoles/length}*2` be extracted to some kind of temporary variable here?